### ISSUE TITLE
get reconstructed mutation catalogues

### ISSUE THREAD

**Issue by MariaCartolano** [kgori/sigfit#48]
Hello,

I have performed the fit_extract_signatures(), selected the desired additional de novo signatures step and now would like to obtain the reconstructed catalogue from my final model: fitted+extracted.

I have used the function get_reconstructions(), but I failed to understand its output.

Could you please help me with this?
Thank you for your help,
Maria

**Comment by baezortega** [comment]
Hi Maria,

I think `get_reconstructions` is an internal function we use within the package to get the MCMC samples of the reconstructed spectra (I think it is by mistake that users can access the function).

You can get point estimates and HPD intervals for the reconstructed spectra using 

`reconstructions <- retrieve_pars(output, "reconstructions")`

with `output` being the object you get from running `fit_extract_signatures()`. The result is a list with three matrices (with same dimensions as your original spectra), corresponding to the mean values of the reconstructed spectra, and the lower and upper ends of their HPD intervals. If what you want is to compare the reconstructed catalogues to the original ones, then the `mean` matrix should be fine. The only issue with this matrix is that it doesn't tell you which proportion of each component in the spectrum comes from each signature (but you can find that out by multiplying signatures and exposures).

You can also plot the reconstructions using `plot_reconstruction()` or `plot_all()`.

Best,
Adrian

**Comment by MariaCartolano** [comment]
Hi Adrian,

thank you for your helpful answer and sorry to come back at this after a while, but I have another question. I went through the code of `retrieve_pars()` and `get_reconstructions()` and I have noticed that while producing the reconstructed counts, you multiply the dot products of the (exposure * signatures) by the number of mutations per sample:

arr <- aperm(sapply(1:NREP, function(i) 
  {
  e$exposures[i, sample, ] * e$signatures[i, , 
                                          ] * sum(counts[sample, ])
}
, simplify = "array"), c(3, 1, 2))

GIven that the formula in its simplest form is  C ≈ E *S, I cannot come up with a straightforward answer. However, it is important for me to know so that I can interpret my results correctly.
Best wishes,
Maria
